sdk/Sdk/FunctionMetadataJsonWriter.cs

```diff
@@ -34,9 +36,26 @@ private static JsonSerializerOptions CreateSerializerOptions()
         public static void WriteMetadata(IEnumerable<SdkFunctionMetadata> functions, string metadataFileDirectory)
         {
             string metadataFile = Path.Combine(metadataFileDirectory, FileName);
-            using var fs = new FileStream(metadataFile, FileMode.Create, FileAccess.Write);
-            using var writer = new Utf8JsonWriter(fs, new JsonWriterOptions { Indented = true });
-            JsonSerializer.Serialize(writer, functions, s_serializerOptions);
+            string newContent = JsonSerializer.Serialize(functions, s_serializerOptions);
+            if (TryReadFile(metadataFile, out string? current) && string.Equals(current, newContent, StringComparison.Ordinal))
```

Comment:
Do we need the other method? Why not inline the file read here?

```
if (File.Exists(metadataFile) && string.Equals(File.ReadAllText(metadataFile), newContent, StringComparison.Ordinal))
```

Reply:
That way works, but having another method isn't hurting anything. I wanted to keep file operations separate from the consumption of the data.